src/Illuminate/Http/Client/PendingRequest.php

```diff
@@ -1035,10 +1036,22 @@ protected function sendRequest(string $method, string $url, array $options = [])
             $this->transferStats = $transferStats;
         };
 
-        return $this->buildClient()->$clientMethod($method, $url, $this->mergeOptions([
+        $mergedOptions = $this->mergeOptions([
             'laravel_data' => $laravelData,
             'on_stats' => $onStats,
-        ], $options));
+        ], $options);
+
+        $mergedOptions = array_map(function($mergedOption) {
+            if ( is_array( $mergedOption ) ) {
+                array_walk_recursive($mergedOption, function (&$value) {
+                    $value = $value instanceof Stringable ? $value->toString() : $value;
+                });
+            }
+
+            return $mergedOption instanceof Stringable ? $mergedOption->toString() : $mergedOption;
+        }, $mergedOptions);
+
+        return $this->buildClient()->$clientMethod($method, $url, $mergedOptions);
```

Comment:
Not a gigantic fan of this implementation.

Super happy for someone to come up with a more elegant approach!

Reply:
There's a build in stringable interface. But then you could just cast anything coming true to string instead of using the method. One caveat would be the boolean bug though.